`lisp-module` build fails when trying to use `cl-charms`

### Describe the bug
`cl-charms` is missing a dependency, hence the build fails.

### Steps To Reproduce
Steps to reproduce the behavior:
1. Add `cl-charms` to `sbcl`:
```nix
  sbcl' = pkgs.sbcl.withPackages (ps: with ps; [
    quick-patch
    cl-charms
  ]);
```
2. Run `nix-build $NIXPKGS -A <new-package>`
3. You end up with this error:
```console
$ nix log /nix/store/av198lzyzhzvary060q99ch87686mks1-cl-charms-20230618-git.drv | cat
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/rx75h995286xsfv01bh90bqwbn589qgb-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
This is SBCL 2.3.7.nixos, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
* #<PACKAGE "NIX-CL-USER">
* #<PACKAGE "NIX-CL-USER">
* T
* LOAD-SYSTEMS
* ; compiling file "/nix/store/rx75h995286xsfv01bh90bqwbn589qgb-source/src/features.lisp" (written 01 JAN 1970 12:00:01 AM):

; wrote /build/source/src/features-tmpGHU3ALSV.fasl
; compilation finished in 0:00:00.005
; compiling file "/nix/store/rx75h995286xsfv01bh90bqwbn589qgb-source/src/low-level/package.lisp" (written 01 JAN 1970 12:00:01 AM):

; wrote /build/source/src/low-level/package-tmpAAURSO1.fasl
; compilation finished in 0:00:00.001
; gcc -o /build/source/src/low-level/curses-grovel__grovel-tmp5GEXGEG5.o -c -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wunused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer -fPIC -I/nix/store/sgd15g1l196r5bp8n7cpyksvbgq56n66-cffi-grovel-20230618-git/ /build/source/src/low-level/curses-grovel__grovel.c
/build/source/src/low-level/curses-grovel__grovel.c:6:10: fatal error: ncurses.h: No such file or directory
    6 | #include <ncurses.h>
      |          ^~~~~~~~~~~
compilation terminated.
; 
; compilation unit aborted
;   caught 1 fatal ERROR condition
BUILD FAILED: Subprocess #<UIOP/LAUNCH-PROGRAM:PROCESS-INFO {1001F415B3}>
 with command ("gcc" "-o"
               "/build/source/src/low-level/curses-grovel__grovel-tmp5GEXGEG5.o"
               "-c" "-g" "-Wall" "-Wundef" "-Wsign-compare" "-Wpointer-arith"
               "-O3" "-D_LARGEFILE_SOURCE" "-D_LARGEFILE64_SOURCE"
               "-D_FILE_OFFSET_BITS=64" "-Wunused-parameter"
               "-fno-omit-frame-pointer" "-momit-leaf-frame-pointer" "-fPIC"
               "-I/nix/store/sgd15g1l196r5bp8n7cpyksvbgq56n66-cffi-grovel-20230618-git/"
               "/build/source/src/low-level/curses-grovel__grovel.c")
 exited with error code 1
```

### Expected behavior
`cl-charms` should build without any issues

### Screenshots
\-

### Additional context
\-

### Notify maintainers
@Uthar 
@7c6f434c 

### Metadata
```console
nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.43, NixOS, 23.05 (Stoat), 23.05.2613.011567f35433`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(ashvith): `""`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/lisp-modules/ql.nix
+++ b/pkgs/development/lisp-modules/ql.nix
@@ -223,7 +223,43 @@ let
       lispLibs = o.lispLibs ++ [
         self.mcclim
       ];
-});
+    });
+    cl-charms = super.cl-charms.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.ncurses ];
+    });
+    libusb-ffi = super.libusb-ffi.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libusb-compat-0_1 ];
+    });
+    cl-fam = super.cl-fam.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.fam ];
+    });
+    jpeg-turbo = super.jpeg-turbo.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libjpeg_turbo ];
+    });
+    vorbisfile-ffi = super.vorbisfile-ffi.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libvorbis ];
+    });
+    png = super.png.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libpng ];
+    });
+    zmq = super.zmq.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.czmq ];
+    });
+    consfigurator = super.consfigurator.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.acl pkgs.libcap ];
+    });
+    cl-gss = super.cl-gss.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libkrb5 ];
+    });
+    magicffi = super.magicffi.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.file ];
+    });
+    keystone = super.keystone.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.keystone ];
+    });
+    capstone = super.capstone.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.capstone ];
+    });
   });
 
   qlpkgs =

